### PR TITLE
fix(relative_url_resolver): Improve relative URI resolution for html

### PR DESCRIPTION
**Pull request by TedSander**
Certain values used in href, src, and action attributes inside of HTML will be rewritten
incorrectly. Such as javascript: values, empty strings, #, and data: values. Add improved
calculation if URIs are relative or not.

### CODE REVIEW

File: test/core_dom/resource_url_resolver_spec.dart [hunk 1]
```diff
@@ -241,6 +241,50 @@ void main() {
...
+  describe('url_resolver uri rewrite', () {
+    ResourceUrlResolver resolver;
+
+    beforeEach((){
+      var config = new ResourceResolverConfig.resolveRelativeUrls(true);
+      var typeMapper = new DynamicTypeToUriMapper();
+      resolver = new ResourceUrlResolver.forTests(typeMapper, config, 'http://localhost');
+    });
+
+    it('should not rewrite absolute URIs ', () {
```
Comment (rkirov) [inline test/core_dom/resource_url_resolver_spec.dart]: technically empty sting is not absolute URI, so I would say `absolute paths, empty paths and custom schemas`.

**Comment by rkirov** [comment]
Add closes #1667 to the commit

**Comment by TedSander** [comment]
Rado. The tests found a problem with the implementation. Fixed the relative URI logic to return the path that conforms with the tests. Can you PTAL?

File: test/core_dom/resource_url_resolver_spec.dart [hunk 2]
```diff
@@ -241,6 +241,60 @@ void main() {
...
+      expect(resolver.combine(baseUri, "test.jpg"))
+          .toEqual("/packages/a.b.c/test.jpg");
+      expect(resolver.combine(baseUri, "test/test.jpg"))
+          .toEqual("/packages/a.b.c/test/test.jpg");
+    });
+
+    it('should rewrite current base uri to absolute without scheme', () {
+      Uri baseUri = Uri.parse("http://localhost/test/test2/test.jgp");
+      expect(resolver.combine(baseUri, 'test4.jpg'))
+          .toEqual('/test/test2/test4.jpg');
```
Comment (rkirov) [inline test/core_dom/resource_url_resolver_spec.dart]: why not leave absolute urls?

**Pull request closed without merging**